azure/scope/identity.go

```diff
@@ -239,7 +239,8 @@ func (p *AzureCredentialsProvider) GetClientSecret(ctx context.Context) (string,
 		if err := p.Client.Get(ctx, key, secret); err != nil {
 			return "", errors.Wrap(err, "Unable to fetch ClientSecret")
 		}
-		return string(secret.Data[AzureSecretKey]), nil
+
+		return string(secret.Data["AZURE_CLIENT_SECRET"]), nil
```

Comment:
I think I may have found a bug here. The secret is stored under `"AZURE_CLIENT_SECRET"`, but `AzureSecretKey` is set to `"clientSecret"`, which returns an empty secret name. @nojnhuh can you confirm if this is a bug or is this intentional?

Reply:
I believe this should be `clientSecret`, at least to stay in sync with this:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/9d2157f31d73e3cacda925df242c8720c262df90/hack/create-identity-secret.sh#L32

I'd be surprised if this were an existing bug given the state of the tests seems green overall.

Reply:
`clientSecret` is correct according to https://capz.sigs.k8s.io/topics/multitenancy#aad-pod-identity-using-service-principal-with-client-password-deprecated